src/bun.js/bindings/ZigGlobalObject.cpp

```diff
@@ -3825,13 +3823,13 @@ void GlobalObject::visitChildrenImpl(JSCell* cell, Visitor& visitor)
     thisObject->m_builtinInternalFunctions.visit(visitor);
     thisObject->m_commonStrings.visit<Visitor>(visitor);
     visitor.append(thisObject->m_assignToStream);
+    visitor.append(thisObject->m_nodeModuleOverriddenResolveFilename);
```

Comment:
☝️ If the structure trick is cheap perf wise it may be a better approach than 1 time trigger.